docs/Project-Configuration/Project-Configuration-Files.md

```diff
@@ -523,28 +516,37 @@ tasks:
           sleep 1000
 ```
 
-By default, a command failing during the `pre` step will not cause the
-entire task to fail. If you want to enforce that failures during `pre`
-cause the task to fail, set the field `pre_error_fails_task` to true.
-Likewise, setting the field `post_error_fails_task` to true will enforce
-that failures in `post` cause the task to fail.
-
-If pre_error_fails_task is set to true and pre fails, it will exit early 
-and not run the main task commands, but it will still run the post block.
-
-If post_error_fails_task is set to true and both the main task and post 
-block fail, the failing command in the main task will be displayed as the 
-failing command, not the failing post command.  The value set for 
-pre_error_fails_task and post_error_fails_task has no effect on tasks run
-in task groups which instead use the settings defined for that task group.
-
 ```yaml
 exec_timeout_secs: 60
 pre_error_fails_task: true
 pre:
   - ...
 ```
 
+#### Special Behavior for Pre, Post, and Timeout
+By default, a command failing during the `pre` block will not cause the entire
+task to fail. If you want to enforce that failures during `pre` cause the task
+to fail, set the field `pre_error_fails_task` to true. If `pre_error_fails_task`
+is set to true and a command in `pre` fails, it will exit early and not run the
+main task commands, but it will still run the `post` block.
+`pre_error_fails_task` has no effect on tasks run in task groups because they do
+not run `pre`; instead, those tasks follow the settings defined for that task
+group.
+
+Similar to `pre`, by default, a command failing during the `post` block will not
+cause the entire task to fail. If you want to enforce that failures during
+`post` cause the task to fail, set the field `post_error_fails_task` to true. If
+`post_error_fails_task` is set to true and both the main task and post block
+have failing commands, the task's failing command will be the command that
+failed in the main task block, not the failing post command.
+`post_error_fails_task` has no effect on tasks run in task groups because they
+do not run `post`; instead, those tasks, follow the settings defined for that
+task group.
+
+By default, commands in `post` or `timeout` will time out after 15 minutes. You
```

Comment:
This is not a silent failure, because the task logs explicitly state that the command fails and on top of that, now shows "Hit <type of timeout> timeout" before the command is failed.